packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,537 @@
+/**
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GitHubAPI } from 'probot/lib/github';
+
+interface CheckParam {}
+
+interface CheckRun {
+  name: string;
+  conclusion: string;
+}
+
+interface CheckStatus {
+  context: string;
+  state: string;
+}
+interface Reviews {
+  user: {
+    login: string;
+  };
+  state: string;
+}
+
+interface ReviewsRequested {
+  users: [];
+  teams: [];
+}
+
+interface PR {
```

Comment:
Can we rename this to `PullRequest`? We generally want to avoid abbreviations in class and variable names.